server/protocols/webadmin/src/test/java/org/apache/james/webadmin/routes/UsersRoutesTest.java

```diff
@@ -0,0 +1,434 @@
+        }
+
+        @Test
+        public void postShouldReturnUserErrorWhenNoBody() {
+            given()
+                .accept(ContentType.JSON)
+                .contentType(ContentType.JSON)
+            .when()
+                .post(UserRoutes.USERS)
+            .then()
+                .statusCode(400);
+        }
+
+        @Test
+        public void postShouldReturnUserErrorWhenEmptyJsonBody() {
+            given()
+                .accept(ContentType.JSON)
+                .contentType(ContentType.JSON)
+                .body("{}")
+            .when()
+                .post(UserRoutes.USERS)
+            .then()
+                .statusCode(400);
+        }
+
+        @Test
+        public void postShouldReturnUserErrorWhenWrongJsonBody() {
+            given()
+                .accept(ContentType.JSON)
+                .contentType(ContentType.JSON)
+                .body("{\"bad\":\"any\"}")
+            .when()
+                .post(UserRoutes.USERS)
+            .then()
+                .statusCode(400);
+        }
+
+        @Test
+        public void postShouldReturnUserErrorWhenMissingPasswordJsonBody() {
+            given()
+                .accept(ContentType.JSON)
+                .contentType(ContentType.JSON)
+                .body("{\"username\":\"username\"}")
+            .when()
+                .post(UserRoutes.USERS)
+            .then()
+                .statusCode(400);
+        }
+
+        @Test
+        public void postShouldReturnUserErrorWhenMissingUserJsonBody() {
+            given()
+                .accept(ContentType.JSON)
+                .contentType(ContentType.JSON)
+                .body("{\"password\":\"password\"}")
+            .when()
+                .post(UserRoutes.USERS)
+            .then()
+                .statusCode(400);
+        }
+
+        @Test
+        public void postShouldReturnOkWhenValidJsonBody() {
+            given()
+                .accept(ContentType.JSON)
+                .contentType(ContentType.JSON)
+                .body("{\"username\":\"username@domain\",\"password\":\"password\"}")
+            .when()
+                .post(UserRoutes.USERS)
+            .then()
+                .statusCode(200);
+        }
+
+        @Test
+        public void postShouldAddTheUser() {
+            given()
+                .accept(ContentType.JSON)
+                .contentType(ContentType.JSON)
+                .body("{\"username\":\"" + USERNAME + "\",\"password\":\"password\"}")
+            .when()
+                .post(UserRoutes.USERS)
+            .then()
+                .statusCode(200);
+
+            given()
+                .accept(ContentType.JSON)
+                .contentType(ContentType.JSON)
+            .when()
+                .get(UserRoutes.USERS)
+            .then()
+                .statusCode(200)
+                .body(equalTo("[\"" + USERNAME + "\"]"));
+        }
+
+        @Test
+        public void postingTwoTimesShouldBeAllowed() {
```

Comment:
Yep, but I think you'll get a race condition if you post two times simultaneously.